source/core/parameters_lagrangian.cc

```diff
@@ -409,44 +433,48 @@ namespace Parameters
           prm.get_integer("inserted number of particles at each time step");
         insertion_frequency = prm.get_integer("insertion frequency");
 
-        std::vector<int> axis_order =
-          convert_string_to_vector<int>(prm, "insertion direction sequence");
-        if (axis_order.size() == 2)
-          axis_order.resize(3);
-
-        direction_sequence.reserve(3);
-        direction_sequence.push_back(axis_order[0]);
-        direction_sequence.push_back(axis_order[1]);
-        direction_sequence.push_back(axis_order[2]);
-
-        const std::vector<std::string> point_coordinates_list(
+        // Clear and add:
+        const std::vector<std::string> clearing_box_point_coordinates_list(
           Utilities::split_string_list(
-            prm.get("insertion box points coordinates"), ":"));
+            prm.get("clearing box points coordinates"), ":"));
 
-        std::vector<double> point_coord_temp_1 = Utilities::string_to_double(
-          Utilities::split_string_list(point_coordinates_list.at(0)));
-        std::vector<double> point_coord_temp_2 = Utilities::string_to_double(
-          Utilities::split_string_list(point_coordinates_list.at(1)));
+        std::vector<double> clearing_point_coord_temp_1 =
+          Utilities::string_to_double(Utilities::split_string_list(
+            clearing_box_point_coordinates_list.at(0)));
+        std::vector<double> clearing_point_coord_temp_2 =
+          Utilities::string_to_double(Utilities::split_string_list(
+            clearing_box_point_coordinates_list.at(1)));
 
-        if (point_coord_temp_1.size() == 2 && point_coord_temp_2.size() == 2)
+        if (clearing_point_coord_temp_1.size() == 2 &&
+            clearing_point_coord_temp_2.size() == 2)
           {
-            point_coord_temp_1.resize(3);
-            point_coord_temp_2.resize(3);
+            clearing_point_coord_temp_1.resize(3);
+            clearing_point_coord_temp_2.resize(3);
```

Comment:
Here you are using resize but assuming that the value at [2] will be zero. This is not a good idea. Set it explicitely.

Reply:
I'm not sure I understand. 

When resize is use,  .at(2) will be equal to zero. This is what we want when dim =2

Reply:
Then you can do if dim and resize with dim == 2, then you give it 0 here already.

Reply:
So I just .push_back(0.); if the size is equal to 2 ?